docs/deployment-ux.md

```diff
@@ -0,0 +1,410 @@
+
+1.  Terminate before replace
+
+    If you're using this deployment method, then Blox will kill
+    `(100 - MinHealthyPercent)%` of tasks, and only launch new tasks to
+    replace as they are successfully terminated. This will ensure that
+    there's never more than one version of the Task running at the same
+    time on the same host, but may result in some downtime for the
+    daemon. This is a good fit for daemons that would fail if there is
+    more than one running per instance, and that can tolerate short
+    periods of unavailability.
+
+    After deployment starts:
+
+    ``` {.shell}
+    $ aws ecs describe-environment-status --environment-name SomeEnvironment/Prod --table
+    REVISION                  STATUS       DESIRED   CURRENT   UP-TO-DATE   AVAILABLE
+    SomeEnvironment/Prod:1    Undeploying  0         5         5            5        
+    SomeEnvironment/Prod:2    Deploying    5         0         0            0        
+    ```
+
+    First task in old revision is terminated:
+
+    ``` {.shell}
+    $ aws ecs describe-environment-status --environment-name SomeEnvironment/Prod --table
+    REVISION                  STATUS       DESIRED   CURRENT   UP-TO-DATE   AVAILABLE
+    SomeEnvironment/Prod:1    Undeploying  0         4         4            4        
+    SomeEnvironment/Prod:2    Deploying    5         0         0            0        
+    ```
+
+    First task in new revision is launched:
+
+    ``` {.shell}
+    $ aws ecs describe-environment-status --environment-name SomeEnvironment/Prod --table
+    REVISION                  STATUS       DESIRED   CURRENT   UP-TO-DATE   AVAILABLE
+    SomeEnvironment/Prod:1    Undeploying  0         4         4            4        
+    SomeEnvironment/Prod:2    Deploying    5         1         1            1        
+    ```
+
+    Deployment progresses in a similar fashion until none of the old
+    revision is left:
+
+    ``` {.shell}
+    $ aws ecs describe-environment-status --environment-name SomeEnvironment/Prod --table
+    REVISION                  STATUS       DESIRED   CURRENT   UP-TO-DATE   AVAILABLE
+    SomeEnvironment/Prod:1    Undeploying  0         0         0            0        
+    SomeEnvironment/Prod:2    Deploying    5         4         4            4        
+    ```
+
+    The final task is launched and the new revision becomes active:
+
+    ``` {.shell}
+    $ aws ecs describe-environment-status --environment-name SomeEnvironment/Prod --table
+    REVISION                  STATUS       DESIRED   CURRENT   UP-TO-DATE   AVAILABLE
+    SomeEnvironment/Prod:1    Inactive     0         0         0            0        
+    SomeEnvironment/Prod:2    Active       5         5         5            5        
+    ```
+
+2.  Terminate after replace
+
+    If you're using this deployment method, then Blox will launch
+    `MaxOver` tasks on as many instances, and only kill old tasks once
+    the new task is confirmed to be running. This will ensure that
```

Comment:
We really want a better signal than "running" to trigger terminating the prior version; the lifecycle state one wants is "ready".  Until we support stronger lifecycle hooks/health-checks, we might get by with a user attribute specifying delay-before-terminate-old-secs.

Reply:
Good point. And if/when we get support for some kind of health check (e.g. aws/amazon-ecs-agent#534), we can use that.